WPMedia/fusion-news-theme-blocks
# PEN-1546: [CMG] Updates to Horizontal Links styling in nav bar

**If you have not filled out the checklist below, the pr is not ready for review.**

## Description
Styling fixes & improvements for nav bar. Fixed search bar styling, horizontal link truncation, other style improvements.

## Jira Ticket
- [PEN-1546](https://arcpublishing.atlassian.net/browse/PEN-1546)

## Acceptance Criteria
1. The horizontal links from  are updated to match the design specifications 
a. Specifically, pay attention to the behavior when the Search bar is expanded – it needs to push other elements to the right. Right now, it does not expand at all on smaller desktop breakpoints. 
b. We should also fix the expanded search bar styling in this ticket – it’s not matching the design specs 
2. The horizontal links should be aligned left within the navigation bar (rather than centered, as they are now)
3. Tests and documentation are updated to cover this functionality 

## Test Steps
1. In `Fusion-News-Theme` repo, checkout `master` & `git pull`
2. After checking out this feature branch in `fusion-news-theme-blocks`, run `rm -rf ./node_modules && npx lerna clean && npm i && npx lerna bootstrap && cd blocks/header-nav-chain-block && npm i && cd ../..`
3. Run `npx fusion start -f -l @wpmedia/header-nav-chain-block,@wpmedia/shared-styles` in `Fusion-News-Theme`
4. Open homepage in PB editor and test three different combinations of settings. After each change, stage & publish page and open http://localhost/pf/homepage/?_website=the-dagen in separate tab:
a. Logo left-aligned **without** horizontal links defined
b. Logo left-aligned **with** horizontal links defined
c. Logo center-aligned (with or without horizontal links defined - shouldn't matter but feel free to test both)
5. Test clicking on search button and search input box should expand (pushing other elements to the right). The search box should have the same height as the other nav bar buttons.
6. Test making the viewport smaller which cuts off space for the horizontal links to display. When a horizontal links bar has too many links to fit on the current viewport width, the horizontal link elements should be truncated as needed (disappear when no space available for them). Horizontal links bar should be left-aligned against the logo now instead of centered within the available space.
7. Ensure that everything displays as expected across breakpoints. When the logo is centered, it should display pretty much the same on tablet/mobile (no horizontal links). When the logo is left-aligned, the horizontal links should be hidden on the mobile breakpoint.

## Effect Of Changes
### Before
Search input bar was too tall (greater height than other nav bar buttons), horizontal links bar was center aligned, horizontal links did not properly truncate items that there was no space for.

### After
Search input bar should now open properly pushing all other elements in the nav bar to the right. Search bar should also display with same height as other nav bar buttons. Horizontal links bar is now left-aligned and properly truncates items that do not have enough space to display fully.

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).
